deploy/kubernetes/helm/che/charts/che-jaeger/templates/service-zipkin.yaml

```diff
@@ -0,0 +1,25 @@
+#
+# Copyright (c) 2012-2018 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: zipkin
+  labels:
+    app: jaeger
+    jaeger-infra: zipkin-service
+spec:
+  ports:
+  - name: jaeger-zipkin-http
+    port: 9411
+    protocol: TCP
+    targetPort: 9411
+  selector:
+    jaeger-infra: jaeger-pod
+  type: ClusterIP
```

Comment:
I took a simplistic approach here and more or less copied the jaeger-all-in-one configuration we have openshift. I am not actually sure much of this is needed at all.

Reply:
See https://github.com/eclipse/che/blob/master/deploy/openshift/templates/jaeger-all-in-one-template.yml

Reply:
Well, I see why it is implemented in this way in Kubernetes Helm Chart =)
I see why it was implemented in this way in OpenShift templates: it was just copied/pasted somewhere from `jaeger`.
But what may be worse than it might be:
1. One more created service - is one more running container that is not really needed. So, may be optimized.
2. As far as I understand from documentation `LoadBalancer` service is used to make pod port's available publicly via using a cloud provider’s load balancer with created needed `NodePort` and `ClusterIP` services https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types. But we don't really use `ClusterIP` service to expose externally because we use Ingresses for that.

I've described points which may be optimized, I guess. But I'm OK to merge the current approach.

Reply:
I've simplified the deployment and the services a bit and the collection seems to be still working. Thanks for the suggestions. What do you say about it now?

Reply:
@metlos We still use three services that expose the same pod, but I'm OK with the current approach since different ports are exposed by these services and different services look like a logical dividing one set of ports from another (agent, collector, query).
Just to note that alternative way - one service jeager with all ports:`agent-zipkin-thrift`, `agent-compact`, `agent-binary`, `agent-sampler-manager`, `jaeger-collector-tchannel`, `jaeger-collector-http`, `query-http`